src/views/Analysis/AnalysisEdit.vue

```diff
@@ -0,0 +1,241 @@
+										:selected="item.id"
+										name="tools"
+										:required="true"
+									/>
+									<a
+										class="absolute text-xl remove-tool-btn cursor-pointer"
+										style="right: -1em"
+										@click="deleteTool(index)"
+										>&times;</a
+									>
+								</div>
+							</div>
+							<UiButton
+								text="+ Pridať nástroj"
+								@click="addTool()"
+								class="rounded-full secondary"
+							/>
+						</div>
+					</UiLabel>
+				</slot>
+
+				<div class="flex flex-row justify-end mt-5">
+					<UiButton
+						text="Uložiť"
+						:disabled="this.loading"
+						:icon="
+							saving
+								? { type: ['fas', 'spinner'], spin: true }
+								: null
+						"
+						type="submit"
+						class="rounded-full w-24 primary"
+					></UiButton>
+				</div>
+			</form>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import UiButton from '@/components/ui/UiButton.vue';
+import UiLabel from '@/components/ui/UiLabel.vue';
+import UiSelect from '@/components/ui/UiSelect.vue';
+import { mapActions, mapGetters } from 'vuex';
+
+export default defineComponent({
+	components: {
+		UiButton,
+		UiLabel,
+		UiSelect
+	},
+	data: () => {
+		return {
+			saving: false,
+			loading: true,
+			users: [
+				{
+					id: 'f7d9caf4-9101-4fe5-859a-f286272640a3',
+					first_name: 'Nikoleta',
+					last_name: 'Hroncova',
+					email: 'nikoleta@gmail.com'
+				},
+				{
+					id: 'f7d9caf4-9101-4fe5-859a-f286272640b3',
+					first_name: 'Petra',
+					last_name: 'Hroncova',
+					email: 'petra@gmail.com'
+				}
+			]
+		};
+	},
+	computed: {
+		...mapGetters('AnalysisStore', ['analysis']),
+		...mapGetters('LabStore', ['labs']),
+		...mapGetters('ToolStore', ['tools']),
+		...mapGetters('SampleStore', ['samples'])
+	},
+	methods: {
+		...mapActions('AnalysisStore', [
+			'fetchAnalysis',
+			'updateAnalysis',
+			'resetState'
+		]),
+		...mapActions('AppStore', ['setAlert']),
+		...mapActions('LabStore', ['fetchLabs']),
+		...mapActions('ToolStore', ['fetchTools']),
+		...mapActions('SampleStore', ['fetchSamples']),
+
+		addTool: function () {
+			this.analysis.tools.push({});
+		},
+		deleteTool: function (index: number) {
+			this.analysis.tools = this.analysis.tools.splice(index, 1); // 2nd parameter means remove one item only
+		},
+		handleSubmit() {
+			this.saving = true;
+			if ((this.$refs.form as HTMLFormElement).checkValidity()) {
+				return this.updateAnalysis({
+					id: this.analysis.id,
+					analysis: this.analysis
+				}).then(() => {
+					this.saving = false;
+					return this.$router.push({
+						name: 'analysis-detail',
+						params: { id: this.analysis.id }
+					});
+				});
+			} else {
+				this.saving = false;
+				(this.$refs.form as HTMLFormElement).reportValidity();
+			}
+		}
+	},
+	mounted: function () {
+		// initial loader
+		this.loading = true;
+		// fetch data from BE
+		this.fetchAnalysis(this.$route.params.id).finally(() => {
```

Comment:
So, only when analysis is fetched, the loader is hidden? What about the other async assets as fetchlabs or fetchTools? 

Please use `Promise.all()` and then disable the loader.